### PR TITLE
ADEN-13542 AE/IE partners logic moved behind document load but setup part done as fast as possible

### PR DESCRIPTION
**Pull request by preludek**

### CODE REVIEW

File: src/platforms/shared/utils/instant-config.setup.ts [hunk 1]
```diff
@@ -32,6 +32,7 @@ export class InstantConfigSetup implements DiProcess {
 			instantConfigEndpoint,
 			instantConfigFallbackEndpoint,
 			instantConfigVariant: context.get('wiki.services_instantConfig_variant'),
+			...(window?.ads?.context?.warp && { lockDelay: 0 }),
```
Comment (jbienkowski-fandom) [inline src/platforms/shared/utils/instant-config.setup.ts:35]:
```suggestion
			...(window?.ads?.context?.warp ? { lockDelay: 0 } : {}),
```

File: src/platforms/shared/setup/ad-engine-phases-setup.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+		adEnginePhases.firstAdCall = waitForEventPromise(
+			eventsRepository.AD_ENGINE_SLOT_LOADED,
+			context.get('options.phases.firstAdCallTimeout') ?? DEFAULT_AD_CALL_WAIT_TIMEOUT,
+		);
+	}
+
+	setupWarpContext(): void {
+		context.set('options.phases', this.instantConfig.get('icWarp', {}));
+		context.set('options.phases.enabled', !!this.instantConfig.get('icWarp'));
```
Comment (jbienkowski-fandom) [inline src/platforms/shared/setup/ad-engine-phases-setup.ts:55]: `window?.ads?.context?.warp ?? false` let's use one var to control both setups, `icWarp` can provide timeout config

File: src/platforms/shared/utils/instant-config.setup.ts (hunk 1 above)
Comment (preludek) [reply src/platforms/shared/utils/instant-config.setup.ts:35]: We don't need ternary here, above should work as expected 🙈

File: src/platforms/shared/setup/ad-engine-phases-setup.ts (hunk 2 above)
Comment (preludek) [reply src/platforms/shared/setup/ad-engine-phases-setup.ts]: done

File: src/platforms/shared/setup/ad-engine-phases-setup.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+		adEnginePhases.firstAdCall = waitForEventPromise(
+			eventsRepository.AD_ENGINE_SLOT_LOADED,
+			context.get('options.phases.firstAdCallTimeout') ?? DEFAULT_AD_CALL_WAIT_TIMEOUT,
+		);
+	}
+
+	setupWarpContext(): void {
+		context.set('options.phases', this.instantConfig.get('icWarp', {}));
+		context.set('options.phases.enabled', window?.ads?.context?.warp);
```
Comment (jbienkowski-fandom) [inline src/platforms/shared/setup/ad-engine-phases-setup.ts:55]:
```suggestion
		context.set('options.phases.enabled', window?.ads?.context?.warp === true);
```
this way we will always deal with boolean (as JS handles logical expressions in a very weird way)

**Review by jbienkowski-fandom — DISMISSED** [review]

File: @types/ads/index.d.ts [hunk 4]
```diff
@@ -42,6 +42,7 @@ interface AdsContext {
 		main_entity_type: string;
 		main_vertical: string;
 	};
+[key: string]: unknown;
```
Comment (jbienkowski-fandom) [inline @types/ads/index.d.ts:45]: where linter?

Comment (preludek) [reply @types/ads/index.d.ts:45]: fixed

**Review by jbienkowski-fandom — APPROVED** [review]

**Pull request merged**